src/StreamFactoryInterface.php

```diff
@@ -7,17 +7,32 @@
 interface StreamFactoryInterface
```

Comment:
@shadowhand I think I answered that in depth [above](https://github.com/http-interop/http-factory/pull/16#issuecomment-247449660).

PSR-7 does not imply streams being backed exclusively by resources - for example, in the PSR, examples such as "socket, pipe, or *callback-based* stream" are used.

There are already many different implemented types of streams in the wild, some of which are not simply wrappers around a stream-resource.

Just to play devil's advocate though, *on the other hand*, if we're *positive* that we are *never* going to support any other type of stream, then I suppose there is not much point in qualifying the name. The main reason I wanted to qualify it, is to avoid ambiguity with other potential future stream-factories for other types of streams - and to avoid the perception that this is interface represents a *general* stream factory, which, if resource-based streams are the only supported type, it does *not*.